creator-node/src/routes/tracks.js

```diff
@@ -58,21 +65,119 @@ module.exports = function (app) {
         return errorResponseBadRequest(req.fileSizeError || req.fileFilterError)
       }
 
-      await AsyncProcessingQueue.addTrackContentUploadTask({
-        logContext: req.logContext,
-        req: {
-          fileName: req.fileName,
-          fileDir: req.fileDir,
-          fileDestination: req.file.destination,
-          session: {
-            cnodeUserUUID: req.session.cnodeUserUUID
+      const isTranscodeQueueAvailable = await TranscodingQueue.isAvailable()
+
+      if (isTranscodeQueueAvailable) {
+        await AsyncProcessingQueue.addTrackContentUploadTask({
+          logContext: req.logContext,
+          req: {
+            fileName: req.fileName,
+            fileDir: req.fileDir,
+            fileDestination: req.file.destination,
+            session: {
+              cnodeUserUUID: req.session.cnodeUserUUID
+            }
           }
-        }
-      })
+        })
+      } else {
+        handleTrackHandOff(
```

Comment:
```suggestion
        handOffTrackTranscodeToAvailableSP(
```

Reply:
as discussed, when we move the decisioning of where to handle track transcode and put it squarely HERE (perhaps in another function) maybe the other function where you house the logic of whether i am available or not goes in a unit-testable util called `canPrimaryHandleTranscodeByItself()`
```js
const primaryCanHandleTranscode = utils.currentNodeCanHandleTranscodeByItself() // sth like this - you can unit test this where it spits out true or false and you mock the parts that matter - initialized is false, or queue is available is false 
if (primaryCanHandleTranscode) { 
    // rest of flow - already tested
} else {
    // maybe wrap this in a fn called handOffTrackTranscode() which just calls
    asyncProcessingQueue.addTranscodeAndSegmentTask()
}
```
oops put this in two places